core/unit_test/openmp/TestOpenMP_Category.hpp

```diff
@@ -50,5 +50,6 @@
 #define TEST_CATEGORY openmp
 #define TEST_CATEGORY_DEATH openmp_DeathTest
 #define TEST_EXECSPACE Kokkos::OpenMP
+#define TEST_CATEGORY_FIXTURE(name) openmp_##name
```

Comment:
Why do we prefer for this thing now openmp_count_bugs.when_all_cycle instead of openmp.count_bugs.when_all_cycle?

Reply:
I don't understand. Is there some way that I'm not aware of to do test fixtures that combines things with a dot like that? We really should be doing all of this with text fixture templates instead of preprocessor macros, but that's a much bigger change.

Reply:
Because fixtures are the correct way to use Google Test when you have common setup/teardown code and common data members, according to the [Google Test documentation](https://github.com/google/googletest/blob/master/googletest/docs/primer.md), and I didn't want to keep doing the wrong thing of copy-pasting setup code all over the place. Is that a problem?

Reply:
And you have to admit, the amount of copy-pasting in the current test suite makes it very difficult to maintain, especially if something in the common setup sequence (of, say, a new feature like this) were to change.

Reply:
From the Google Test documentation:

> Tests should be well organized and reflect the structure of the tested code. googletest groups related tests into test suites that can share data and subroutines. This common pattern is easy to recognize and makes tests easy to maintain.

Reply:
Ah I overlooked that you are actually using a fixture with shared code and a SetUp call ... 
Makes sense now.